src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1395,12 +1452,25 @@ private void ExtendWindow(int amount)
         private void Abort(Exception abortException)
         {
             // The connection has failed, e.g. failed IO or a connection-level frame error.
-            if (Interlocked.CompareExchange(ref _abortException, abortException, null) != null &&
-                NetEventSource.Log.IsEnabled() &&
-                !ReferenceEquals(_abortException, abortException))
+            lock (SyncObject)
             {
-                // Lost the race to set the field to another exception, so just trace this one.
-                Trace($"{nameof(abortException)}=={abortException}");
+                if (_abortException == null)
```

Comment:
Nit: Since you're using `is not null` elsewhere:
```suggestion
                if (_abortException is null)
```